assembly/src/ast/module.rs

```diff
@@ -185,6 +185,11 @@ impl ModuleAst {
         self.reexported_procs.write_into(target);
         target.write_u16(self.local_procs.len() as u16);
         self.local_procs.write_into(target);
+
+        // serialize source locations if required
+        if options.serialize_source_locations {
+            self.write_source_locations(target);
```

Comment:
Should we get rid of `write_source_locations` method and move the code here? Is there a reason we'd need it after these chagnes?